src/scripts/main.js

```diff
@@ -1,3 +1,296 @@
+  });
+
+  if (e.target.classList.contains('asc')) {
+    e.target.classList.remove('asc');
+    tbody.append(...sort.reverse());
+  } else {
+    e.target.classList.add('asc');
+    tbody.append(...sort);
+  }
+});
+
+// When clicks on a row, it becomes selected
+
+tbody.addEventListener('click', function(e) {
+  const row = e.target.closest('tr');
+
+  if (e.ctrlKey || e.metaKey) {
+    toggleSelect(row);
+  } else {
+    singleSelect(row);
+  }
+});
+
+function toggleSelect(tr) {
+  tr.classList.toggle('active');
+}
+
+function singleSelect(tr) {
+  const selected = document.querySelectorAll('.active');
+
+  for (const elem of selected) {
+    elem.classList.remove('active');
+  }
+  tr.classList.add('active');
+}
+
+// Added a form to the document
+// Form allows users to add new employees to the spreadsheet
+
+const form = document.createElement('form');
+
+form.classList.add('new-employee-form');
+
+headers.forEach(head => {
+  const label = document.createElement('label');
+
+  label.textContent = head.textContent;
+
+  if (head.textContent === 'Office') {
+    const select = document.createElement('select');
+    const options = [`Tokyo`, `Singapore`,
+      `London`, `New York`, `Edinburgh`, `San Francisco`];
+
+    options.forEach(option => {
+      const city = document.createElement('option');
+
+      city.textContent = option;
+      select.append(city);
+    });
+
+    select.name = head.textContent.toLowerCase();
+    select.dataset.qa = head.textContent.toLowerCase();
+
+    label.append(select);
+  } else {
+    const input = document.createElement('input');
+
+    if (head.textContent === 'Age' || head.textContent === 'Salary') {
+      input.type = 'number';
+    }
+
+    if (head.textContent === 'Salary') {
+      input.step = 1000;
+    }
+
+    input.name = head.textContent.toLowerCase();
+    input.dataset.qa = head.textContent.toLowerCase();
+
+    label.append(input);
+  }
+
+  form.append(label);
+});
+
+const submit = document.createElement('button');
+
+submit.textContent = `Save to table`;
+
+form.append(submit);
+
+document.body.append(form);
+
+// Validation of data in form
+
+submit.addEventListener('click', function(e) {
+  e.preventDefault();
+
+  const data = new FormData(form);
+  const dataObject = Object.fromEntries(data.entries());
+  let withoutMistakes = false;
+
+  if (dataObject.name.length < 4) {
+    pushNotification('error', 'Error', 'Name could not be less than 4 letters');
+  } else if (dataObject.age < 18 || dataObject.age > 90) {
+    pushNotification('error', 'Error', 'Age must be between 18 to 90');
+  } else if (!dataObject.position || dataObject.position.length < 2) {
+    pushNotification('error', 'Error', 'Enter valid position');
+  } else if (!dataObject.salary || isNaN(dataObject.salary)) {
+    pushNotification('error', 'Error', 'Enter valid salary');
+  } else {
+    pushNotification('success', 'Success', 'New employee was added');
+    withoutMistakes = true;
+  }
```

Comment:
You can use sth like this to avoid else if nesting (it's hard to read)

``` 
const validations = {
    name: {
      isValid: dataObject.name.length >= 4,
      message: 'Name could not be less than 4 letters'
    },
    age: {
      isValid: dataObject.age >= 18 && dataObject.age <= 90,
      message: 'Age must be between 18 to 90'
    },
    position: {
      isValid: dataObject.position && dataObject.position.length >= 2,
      message: 'Enter valid position'
    },
    salary: {
      isValid: !isNaN(dataObject.salary) && dataObject.salary > 0,
      message: 'Enter valid salary'
    }
  };

  Object.keys(validations).forEach(key => {
    const validation = validations[key];

    if (!validation.isValid) {
      pushNotification('error', 'Error', validation.message);
      withoutMistakes = false;
    }
  })
  ```